src/storage/kv.js

```diff
@@ -0,0 +1,184 @@
+const api = require('../api');
+const { getNamespaceByName } = require('./storage');
+
+/**
+ * Gets namespaces associated with the account
+ * @param {*} options
+ */
+async function getKeys({ accountId, nameSpace } = {}) {
+  const account = accountId || process.env.CLOUDFLARE_ACCOUNT_ID;
+
+  if (!account) {
+    throw 'You must provide an account ID';
+  }
+
+  const nameSpaceId = await getNamespaceByName({
+    accountId: account,
+    name: nameSpace
+  });
+
+  if (!nameSpaceId) {
+    throw 'Name space does not exist';
+  }
+
+  let result = await api.cfApiCall({
+    url: `/accounts/${account}/storage/kv/namespaces/${nameSpaceId}/keys`,
+    method: 'GET'
+  });
+
+  if (result.success && result.result) {
+    result = result.result;
+  }
+
+  return result;
+}
+
+/**
+ * Gets namespaces associated with the account
+ * @param {*} options
+ */
+async function getAllKeysAndValues({ accountId, nameSpace } = {}) {
+  const account = accountId || process.env.CLOUDFLARE_ACCOUNT_ID;
+
+  if (!account) {
+    throw 'You must provide an account ID';
+  }
+
+  const keys = await getKeys({
+    accountId: account,
+    nameSpace: nameSpace
+  });
+
+  const result = [];
+  for (const key of keys) {
+    const value = await getKeyValue({
+      accountId: account,
+      nameSpace: nameSpace,
+      key: key.name
+    });
+    result.push({ key: key.name, value }).result;
+  }
+
+  return result;
+}
+
+/**
+ * Gets namespaces associated with the account
+ * @param {*} options
+ */
+async function getKeyValue({ accountId, nameSpace, key } = {}) {
+  const account = accountId || process.env.CLOUDFLARE_ACCOUNT_ID;
+
+  if (!account) {
+    throw 'You must provide an account ID';
+  }
+
+  if (!key) {
+    throw 'You must provide a key';
+  }
+
+  const nameSpaceId = await getNamespaceByName({
+    accountId: account,
+    name: nameSpace
+  });
+
+  if (!nameSpaceId) {
+    throw 'Name space does not exist';
+  }
+
+  let result = await api.cfApiCall({
+    url: `/accounts/${account}/storage/kv/namespaces/${nameSpaceId}/values/${key}`,
+    method: 'GET'
+  });
+
+  if (result.success && result.result) {
+    result = result.result;
+  }
+
+  return result;
+}
+
+/**
+ * Gets namespaces associated with the account
+ * @param {*} options
+ */
+async function setKeyValue({ accountId, nameSpace, key, value } = {}) {
+  const account = accountId || process.env.CLOUDFLARE_ACCOUNT_ID;
+
+  if (!account) {
+    throw 'You must provide an account ID';
+  }
+
+  if (!key) {
+    throw 'You must provide a key';
+  }
+
+  if (!value) {
+    throw 'You must provide a value';
+  }
+
+  const nameSpaceId = await getNamespaceByName({
```

Comment:
Same here. These methods should take an optional namespaceId in case the developer already has it or wants to fetch it once. This design could lead to lots of network calls.